app/controllers/vm_cloud_controller.rb

```diff
@@ -21,7 +21,11 @@ def resize
       ) unless @explorer
       @flavors = {}
       unless @record.ext_management_system.nil?
-        @record.ext_management_system.flavors.each { |f| @flavors[f.name] = f.id unless f == @record.flavor }
+        @record.ext_management_system.flavors.each do |f|
+          unless f == @record.flavor || @record.flavor.root_disk_size > f.root_disk_size
```

Comment:
This `unless` is bending my brain.

Maybe:

```ruby
if f != @record.flavor && f.root_disk_size >= @record.flavor.root_disk_size
```

That seems to more clearly say:

"If `this flavor` is not my record's flavor and `this flavor`'s root_disk_size is greater than or equal to my record's flavor's root_disk_size, then include `this flavor`"